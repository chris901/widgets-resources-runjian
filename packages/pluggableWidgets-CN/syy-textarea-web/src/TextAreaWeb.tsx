import { ReactNode, createElement, useCallback } from "react";
import { TextAreaWebContainerProps } from "../typings/TextAreaWebProps";
import { TextAreaComponent } from "./components/TextAreaWeb";
import { executeAction } from "@mendix/piw-utils-internal";
import "./ui/TextAreaWeb.scss";

export function TextAreaWeb(props: TextAreaWebContainerProps): ReactNode {
    const { placeholder, showCount, autoSize, bordered, allowClear, maxLength, minRows, maxRows, value } = props;

    const onChange = useCallback(() => executeAction(props.onChange), [props.onChange]);
    const onResize = useCallback(() => executeAction(props.onResize), [props.onResize]);
    const onPressEnter = useCallback(() => executeAction(props.onPressEnter), [props.onPressEnter]);

    return (
        <div tabIndex={props.tabIndex}>
            <TextAreaComponent
                value={value?.value + ""}
                maxLength={maxLength}
                allowClear={allowClear}
                showCount={showCount}
                autoSize={autoSize === true ? { minRows, maxRows } : false}
                bordered={bordered}
                placeholder={placeholder}
                onChange={onChange}
                onResize={onResize}
                onPressEnter={onPressEnter}
            />
        </div>
    );
}
