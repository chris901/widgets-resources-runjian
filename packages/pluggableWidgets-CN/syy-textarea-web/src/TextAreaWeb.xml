<?xml version="1.0" encoding="utf-8" ?>
<widget id="runjian.textareaweb.TextAreaWeb" pluginWidget="true" needsEntityContext="true" offlineCapable="true" supportedPlatform="Web" xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../node_modules/mendix/custom_widget.xsd">
    <name>SYY TextArea Web</name>
    <description>My widget description</description>
    <icon />
    <properties>
        <propertyGroup caption="General">
            <systemProperty key="Label" />
            <property key="placeholder" type="string" required="false" defaultValue="please enter">
                <caption>placeholder</caption>
                <description>Placeholder</description>
            </property>
            <property key="allowClear" type="boolean" defaultValue="false">
                <caption>allowClear</caption>
                <description>If allow to remove input content with clear icon</description>
            </property>
            <property key="bordered" type="boolean" defaultValue="true">
                <caption>bordered</caption>
                <description>Whether has border style</description>
            </property>
            <property key="value" type="attribute">
                <caption>value</caption>
                <description>The input content value</description>
                <attributeTypes>
                    <attributeType name="String" />
                    <attributeType name="Decimal" />
                    <attributeType name="Integer" />
                </attributeTypes>
            </property>
            <property key="maxLength" type="integer" defaultValue="100">
                <caption>maxLength</caption>
                <description>The max length</description>
            </property>
            <property key="showCount" type="boolean" defaultValue="false">
                <caption>showCount</caption>
                <description>Whether show text count</description>
            </property>
            <property key="autoSize" type="boolean" defaultValue="false">
                <caption>autoSize</caption>
                <description>Height autosize feature, can be set to true | false</description>
            </property>
            <property key="minRows" type="integer" defaultValue="2">
                <caption>minRows</caption>
                <description>Minimum height, at least a few lines</description>
            </property>
            <property key="maxRows" type="integer" defaultValue="6">
                <caption>maxRows</caption>
                <description>Maximum height, up to several lines</description>
            </property>
        </propertyGroup>
        <propertyGroup caption="Events">
            <property key="onChange" type="action">
                <caption>On change</caption>
                <description />
            </property>
            <property key="onResize" type="action">
                <caption>onResize</caption>
                <description />
            </property>
            <property key="onPressEnter" type="action">
                <caption>onPressEnter</caption>
                <description />
            </property>
        </propertyGroup>
    </properties>
</widget>
