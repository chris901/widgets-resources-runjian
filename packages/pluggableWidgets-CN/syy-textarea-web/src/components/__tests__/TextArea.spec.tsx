import { createElement } from "react";
import { render } from "enzyme";
import { TextAreaComponent, TextAreaProps } from "../TextAreaWeb";

describe("TextAreaWeb", () => {
    let defaultTextAreaProps: TextAreaProps;

    beforeEach(() => {
        defaultTextAreaProps = {
            value: "hello"
        };
    });

    // <TextArea
    //             value={value?.value + ""}
    //             maxLength={maxLength}
    //             allowClear={allowClear}
    //             showCount={showCount}
    //             autoSize={autoSize === true ? { minRows, maxRows } : false}
    //             bordered={bordered}
    //             placeholder={placeholder}
    //             onChange={onChange}
    //             onResize={onResize}
    //             onPressEnter={onPressEnter}
    //         />

    it("renders value", () => {
        const textAreaWrapper = render(<TextAreaComponent {...defaultTextAreaProps} value="hello render hello" />);

        expect(textAreaWrapper).toMatchSnapshot();
    });

    it("render value max than maxLength", () => {
        const textAreaWrapper = render(
            <TextAreaComponent {...defaultTextAreaProps} value="hello render hello" maxLength={5} />
        );

        expect(textAreaWrapper.prop("textContent")).toEqual("hello");
    });

    // it("renders a default icon", () => {
    //     const iconWrapper = render(<Icon {...defaultIconProps} data={undefined} />);

    //     expect(iconWrapper).toMatchSnapshot();
    // });

    // it("doesn't render an icon with an unknown icon data type", () => {
    //     const iconWrapper = render(<Icon {...defaultIconProps} data={{ type: "unknown" } as any} />);

    //     expect(iconWrapper).toMatchSnapshot();
    // });

    // it("doesn't apply an animate class to a glyph icon when animate is false", () => {
    //     const iconWrapper = mount(<Icon {...defaultIconProps} animate={false} />);

    //     expect(iconWrapper.find("span").prop("className")).not.toContain("animate");
    // });

    // it("doesn't apply an animate class to an image icon when animate is false", () => {
    //     const iconWrapper = mount(
    //         <Icon {...defaultIconProps} data={{ type: "image", iconUrl: "icon.url" }} animate={false} />
    //     );

    //     expect(iconWrapper.find("img").prop("className")).not.toContain("animate");
    // });

    // it("doesn't apply an animate class to a default icon when animate is false", () => {
    //     const iconWrapper = mount(<Icon {...defaultIconProps} data={undefined} animate={false} />);
    //     expect(iconWrapper.find("div").prop("className")).not.toContain("animate");
    // });
});
