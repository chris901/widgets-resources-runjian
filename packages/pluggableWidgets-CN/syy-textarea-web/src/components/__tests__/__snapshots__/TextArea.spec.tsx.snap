// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaWeb render value max than maxLength 1`] = `
<textarea
  class="ant-input"
>
  hello render hello
</textarea>
`;

exports[`TextAreaWeb renders value 1`] = `
<textarea
  class="ant-input"
>
  hello render hello
</textarea>
`;
